public/js/buttonFunctions.js

```diff
@@ -19,6 +19,29 @@ let newAssignment = function() {
   }
 }
 
+function replaceAssignmentFromID(oldData, newData, classID) {
+  const assignments = currentTableData.currentTermData.classes[classID].assignments;
+
+  //this weird version of indexOf is necessary (I think) because when comparing the two it doesn't always match correctly
+  let index = -1;
+  for (i=0; i<assignments.length-1; i++) {
+    if (assignments[i]["assignment_id"] === oldData["assignment_id"]) {
+      index = i;
+      break;
+    }
+  }
+  assignments[index] = newData;
+  updateGradePage();
+}
+
+function removeAssignmentFromID(id, classID) {
+  const assignments = currentTableData.currentTermData.classes[classID].assignments;
+  const newArray = assignments.filter((obj) => {
+    return obj["assignment_id"] !== id;
+  });
```

Comment:
This arrow function can be simplified:
```suggestion
  const newArray = assignments.filter(({ assignment_id }) =>
    assignment_id !== id
  );
```